src/utils/localstorage.ts

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+
+export function getLocalStorageItem<T>(key: string): T | null {
+  const item = localStorage.getItem(key);
+
+  if (!item) {
+    return null;
+  }
+
+  return JSON.parse(item) as T;
+}
+
+// shoutout to my boy josh comeau for this hook template.
+// https://www.joshwcomeau.com/react/persisting-react-state-in-localstorage/
+// also this for handling nextjs SSR
+// https://upmostly.com/next-js/using-localstorage-in-next-js
+export function useStickyState<T>(
```

Comment:
wait if we're using supabase now, should we just get rid of ydoc? we don't need it for anything except cursors right

Reply:
hmm well right now its persisting the letter positions locally which is better than globally -- i think we should keep it and add on other real-time interactions on the letters too

Reply:
yeah ydoc only tracks # of letter opens, is this something we still want to do

Reply:
oh you mean the stuff not related to the positioing? then we dont need it, although it would be nice to real-time stream that info too